dashboard/ltiv1p3.py

```diff
@@ -0,0 +1,203 @@
+import logging, string, random
+
+import django.contrib.auth
+
+from django.http import JsonResponse
+from django.shortcuts import render
+from django.views.decorators.http import require_POST
+from django.views.decorators.csrf import csrf_exempt
+from pylti1p3.contrib.django import DjangoOIDCLogin, DjangoMessageLaunch, DjangoCacheDataStorage
+from pylti1p3.tool_config import ToolConfDict
+from django.contrib.auth.models import User
+from dashboard.common.db_util import canvas_id_to_incremented_id
+from django.core.exceptions import ObjectDoesNotExist
+from dashboard.models import Course, CourseViewOption
+from django.conf import settings
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_tool_conf():
+    ltiv1_p3 = settings.LTI_CONFIG
+    try:
+        tool_config = ToolConfDict(ltiv1_p3)
+    except Exception as e:
+        raise e
+    # There should be one key per platform and the name relay on platforms generic domain not institution specific
+    platform_domain = list(ltiv1_p3.keys())[0]
+    client_id = ltiv1_p3[platform_domain][0]['client_id']
+    private_key_file_path = ltiv1_p3[platform_domain][0]['private_key_file']
+    public_key_file_path = ltiv1_p3[platform_domain][0]['public_key_file']
+    public_key = public_key_file_path if public_key_file_path else '/secret/public.key'
+    private_key = private_key_file_path if private_key_file_path else '/secret/private.key'
+    try:
+        private_key_content = open(private_key, 'r').read()
+        public_key_content = open(public_key, 'r').read()
+    except OSError as e:
+        raise e
+    tool_config.set_public_key(platform_domain, public_key_content, client_id=client_id)
+    tool_config.set_private_key(platform_domain, private_key_content, client_id=client_id)
+    return tool_config
+
+
+def get_jwks(request):
+    try:
+        tool_conf = get_tool_conf()
+    except (OSError, Exception) as e:
+        error_message = {
+            'lti_error_message': f'Fetching JWKS failed due to {e}'
+        }
+        return JsonResponse(error_message)
+    return JsonResponse(tool_conf.get_jwks())
+
+
+def check_if_instructor(roles, username, course_id):
+    teaching_roles = [role for role in roles if 'membership#Instructor' in role or
+                      'membership/Instructor#TeachingAssistant' in role]
+    if len(teaching_roles) == 0:
+        logger.info(f"user {username} is not instructor in course {course_id}")
+        return False
+
+    ta = [teach_role for teach_role in teaching_roles if 'membership/Instructor#TeachingAssistant' in teach_role]
+    if len(ta) != 0:
+        logger.info(f'user {username} is not an instructor of course {course_id}')
+        return False
+    else:
+        logger.info(f'user {username} is an instructor in course {course_id}')
+        return True
```

Comment:
The logic doesn't seem right to me.  Doesn't Canvas allow users to be instructor and TA for a course at the same time?  If so, this function returns `False` for anyone who has both roles.

Since this function is named `check_if_instructor`, it should look for the explicit instructor role string in their list of roles and return a Boolean value for whether it was found.